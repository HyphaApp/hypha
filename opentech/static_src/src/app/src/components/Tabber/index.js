import React from 'react';

const Tabber = () => <div>Tabber</div>

export default Tabber;
