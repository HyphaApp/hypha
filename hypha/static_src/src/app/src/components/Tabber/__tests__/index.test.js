import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import Tabber from "../index";
import {Tab} from '@components/Tabber'
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });

describe("Test Tabber component", () => {
        
    const children = [
        <Tab button="Status" key="status">
          {/* <span>inside status</span> */}
          </Tab>,
        <Tab button="Notes" key="note"><span>inside note</span></Tab>
    ]

    const subject = mount(<Tabber>{children}</Tabber>
    );
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('.is-active').text()).toBe("Status");
      expect(subject.find('a').length).toBe(2);
      expect(subject.find('a').last().text()).toBe("Notes");
      subject.find('a').last().props().onClick()
    });

    it("Check methods of the component", () => {
        const spy = jest.spyOn(Tabber.prototype, 'componentDidUpdate');
        const wrapper = shallow(<Tabber children={children}/>).instance();
        wrapper.handleClick(1);
        expect(spy).toHaveBeenCalled();
        spy.mockReset();
        spy.mockRestore();
        expect(wrapper.state.activeTab).toBe(1);
        
      });

      it("Check handleClick method  of the component", () => {
          const wrapper = shallow(<Tabber children={children}/>).instance();
          wrapper.handleClick = jest.fn()
          wrapper.handleClick(0);
          expect(wrapper.handleClick).toHaveBeenCalled();
          expect(wrapper.handleClick).toHaveBeenCalledWith(0);
          expect(wrapper.state.activeTab).toBe(0);
        });
    
  
    test('render a Tabber component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

  describe("Test Tabber component2", () => {
        
    const children = [
        <Tab button="Status" key="status">
          <span>inside status</span>
          </Tab>,
        <Tab button="Notes" key="note"><span>inside note</span></Tab>
    ]

    const subject = mount(<Tabber>{children}</Tabber>
    );
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('.is-active').text()).toBe("Status");
      expect(subject.find('a').length).toBe(2);
      expect(subject.find('a').last().text()).toBe("Notes");
      subject.find('a').last().props().onClick()
    });

    it("Check methods of the component", () => {
        const spy = jest.spyOn(Tabber.prototype, 'componentDidUpdate');
        const wrapper = shallow(<Tabber children={children}/>).instance();
        wrapper.handleClick(1);
        expect(spy).toHaveBeenCalled();
        spy.mockReset();
        spy.mockRestore();
        expect(wrapper.state.activeTab).toBe(1);
        
      });

      it("Check handleClick method  of the component", () => {
          const wrapper = shallow(<Tabber children={children}/>).instance();
          wrapper.handleClick = jest.fn()
          wrapper.handleClick(0);
          expect(wrapper.handleClick).toHaveBeenCalled();
          expect(wrapper.handleClick).toHaveBeenCalledWith(0);
          expect(wrapper.state.activeTab).toBe(0);
        });
    
  
    test('render a Tabber component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });
