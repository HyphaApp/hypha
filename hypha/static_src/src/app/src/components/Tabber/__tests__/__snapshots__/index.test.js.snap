// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tabber component render a Tabber component 1`] = `
<Tabber>
  <div
    className="tabber"
  >
    <div
      className="tabber__navigation"
    >
      <a
        className="tabber__link is-active"
        key="status"
        onClick={[Function]}
      >
        Status
      </a>
      <a
        className="tabber__link "
        key="note"
        onClick={[Function]}
      >
        Notes
      </a>
    </div>
    <Tab
      button="Status"
      key="status"
    >
      <div
        className="tab"
      />
    </Tab>
  </div>
</Tabber>
`;

exports[`Test Tabber component2 render a Tabber component 1`] = `
<Tabber>
  <div
    className="tabber"
  >
    <div
      className="tabber__navigation"
    >
      <a
        className="tabber__link is-active"
        key="status"
        onClick={[Function]}
      >
        Status
      </a>
      <a
        className="tabber__link "
        key="note"
        onClick={[Function]}
      >
        Notes
      </a>
    </div>
    <Tab
      button="Status"
      key="status"
    >
      <div
        className="tab"
      >
        <span>
          inside status
        </span>
      </div>
    </Tab>
  </div>
</Tabber>
`;
